homeassistant/components/yeelight/light.py

```diff
@@ -483,6 +496,8 @@ def set_effect(self, effect) -> None:
             except BulbException as ex:
                 _LOGGER.error("Unable to set effect: %s", ex)
 
+            self.device.update()
```

Comment:
Why is this needed ? and wasn't needed before ?

Reply:
Its just to update state right away, after starting effect, so that icon will also change. Before that, there was no real reason to update device state, after starting effect ( flow ). There was no indication in UI, that something is happening with the bulb ( effect is in progress ). Now icon might change, so lets make update asap to indicate that. When you start effect, icon changes right away.